CMakeLists.txt

```diff
@@ -0,0 +1,179 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(opus LANGUAGES C)
+
+include(opus_config.cmake)
+include(opus_sources.cmake)
+include(opus_functions.cmake)
+include(GNUInstallDirs)
+
+add_library(opus ${opus_sources} ${opus_sources_float})
+
+set_target_properties(opus
+                      PROPERTIES SOVERSION
+                                 ${OPUS_LIBRARY_VERSION_MAJOR}
+                                 VERSION
+                                 ${OPUS_LIBRARY_VERSION})
+
+target_include_directories(
+  opus
+  PUBLIC $<BUILD_INTERFACE: ${CMAKE_CURRENT_SOURCE_DIR}/include>
+         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  PRIVATE $<BUILD_INTERFACE:
+          ${CMAKE_CURRENT_SOURCE_DIR}/include
+          ${CMAKE_CURRENT_SOURCE_DIR}/celt
+          ${CMAKE_CURRENT_SOURCE_DIR}/silk>)
+
+target_link_libraries(opus PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+
+target_compile_definitions(opus PRIVATE CUSTOM_MODES OPUS_BUILD)
+
+set_property(TARGET opus PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
Remove it. It was required by object library. CMake will set it automatically for shared library.